src/com/owncloud/android/files/services/AvailableOfflineSyncJobService.java

```diff
@@ -0,0 +1,189 @@
+            for (Pair<OCFile, String> fileForAccount : availableOfflineFilesForAccount) {
+
+                String localPath = fileForAccount.first.getStoragePath();
+
+                if (localPath == null) {
+                    localPath = FileStorageUtils.getDefaultSavePathFor(
+                            fileForAccount.second,
+                            fileForAccount.first
+                    );
+                }
+
+                File localFile = new File(localPath);
+
+                if (localFile.lastModified() <= fileForAccount.first.getLastSyncDateForData()) {
+                    Log_OC.i(TAG, "File " + localPath + " already synchronized, ignoring");
+                    continue;
+                }
+
+                startSyncOperation(fileForAccount.first, fileForAccount.second);
+            }
+        }
+
+        /**
+         * Triggers an operation to synchronize the contents of a recently modified available offline file with
+         * its remote counterpart in the associated ownCloud account.
+         * @param availableOfflineFile file to synchronize
+         * @param accountName account to synchronize the available offline file with
+         */
+        private void startSyncOperation(OCFile availableOfflineFile, String accountName) {
+            Log_OC.i(
+                    TAG,
+                    String.format(
+                            "Requested synchronization for file %1s",
+                            availableOfflineFile.getFileName()
+                    )
+            );
+
+            Account mAccount = AccountUtils.getOwnCloudAccountByName(mAvailableOfflineJobService, accountName);
+
+            FileDataStorageManager storageManager =
+                    new FileDataStorageManager(
+                            mAvailableOfflineJobService, mAccount, mAvailableOfflineJobService.getContentResolver()
+                    );
+
+            SynchronizeFileOperation synchronizeFileOperation =
+                    new SynchronizeFileOperation(availableOfflineFile, null, mAccount, false,
+                            mAvailableOfflineJobService);
+
+            RemoteOperationResult result = synchronizeFileOperation.
+                    execute(storageManager, mAvailableOfflineJobService);
+
+            if (result.getCode() == RemoteOperationResult.ResultCode.SYNC_CONFLICT) {
+                // If the user is not running the app, this can be very intrusive so show a notification to solve
+                // the conflicts
+                Intent i = new Intent(mAvailableOfflineJobService, ConflictsResolveActivity.class);
+                i.setFlags(i.getFlags() | Intent.FLAG_ACTIVITY_NEW_TASK);
+                i.putExtra(ConflictsResolveActivity.EXTRA_FILE, availableOfflineFile);
+                i.putExtra(ConflictsResolveActivity.EXTRA_ACCOUNT, mAccount);
+                mAvailableOfflineJobService.startActivity(i);
```

Comment:
Launching an activity from job service....

This is not a good approach. You know Android restricts starting UI items from services. I'd move to another logic. For example, this job emits an event to start activity

Reply:
> I'd move to another logic. For example, this job emits an event to start activity

Well, using a broadcast in this case would not be that trivial because I cannot register the receiver in `ConflictsResolveActivity` since `onCreate()` method is never called before the job detects conflicts, that activity is only started with the code above. 

Besides, I could register the receiver in other place, but how would I unregister it? In an activity `onDestroy()` method would not work because as soon as you kill the app, the broadcast from the job will not work.

> You know Android restricts starting UI items from services

Is there any official Android restrictions about this? I mean, I agree with you this is not the best approach, but think is not because of Android restrictions, the Android  framework allows you to start an Activity from a Service, but because we are interrupting what the user is currently doing, so the best approach would be to use a Notification instead and once the user decides to solve the conflict, open the activity.